DTLOR_DP.py

```diff
@@ -107,313 +126,221 @@ def DP(hostTree, parasiteTree, phi, locus_map, D, T, L, Origin, R):
-                    oBest[(vp, eh2, top_is_star, l_bottom)]!= [(None, None, None, None)]:
-                        bestSwitchLocations[(vp, eh1, l_bottom)].extend\
-                        (oBest[(vp, eh2, top_is_star, l_bottom)])
-                    if bestSwitch[(ep, eh2, l_bottom)] == ep_bestSwitch and \
-                    bestSwitchLocations[(vp, vh,  l_bottom)] != [(None, None, None, None)]:
-                        bestSwitchLocations[(vp, eh2, l_bottom)].extend\
-                        (bestSwitchLocations[(vp, vh, l_bottom)])
-                    if bestSwitch[(ep, eh2, l_bottom)] == O_eh1 and \
-                    oBest[(vp, eh1, top_is_star, l_bottom)]!=[(None, None, None, None)]:
-                        bestSwitchLocations[(vp, eh2, l_bottom)].extend\
-                        (oBest[(vp, eh1, top_is_star, l_bottom)])
-        #TODO: Combine
-        #TODO: Events
-        left_cost = Infinity
-        right_cost = Infinity
-        # One child gets a synteny
-        for eh in postorder(hostTree, "hTop"):
-            for l in allsynteny:
-                # Left child gets a synteny
-                l_cost = null[ep1] + C[(ep2, eh, l)] + Origin
-                # Right child gets a synteny
-                r_cost = null[ep2] + C[(ep1, eh, l)] + Origin
-                if left_cost > l_cost:
-                    left_cost = l_cost
-                if right_cost > r_cost:
-                    right_cost = r_cost
-        # Neither child gets a synteny
-        both_null = null[ep1] + null[ep2]
-        # Both children get syntenies
-        both_cost
-        for eh1 in postorder(hostTree, "hTop"):
-            for eh2 in postorder(hostTree, "hTop"):
-                for l1 in allsynteny:
-                    for l2 in allsynteny:
-                        b_cost = C[(ep1, eh1, l1)] + C[(ep2, eh2, l2)] + 2 * Origin
-        null[g] = min(left_cost, right_cost, both_null, both_cost)
-
-    min_cost = null["pTop"]
-    #TODO: Combine
-    # Cost for assigning the root an origin
-    for eh in postorder(hostTree, "hTop"):
-        for l in allsynteny:
-            m_cost = C[("pTop", eh, l)] + Origin
-            if min_cost > m_cost:
-                min_cost = m_cost
-
-    #TODO:
-    return min_cost
-
-    for key in bestSwitchLocations:
-        if bestSwitchLocations[key][0] == (None, None, None, None):
-            bestSwitchLocations[key] = bestSwitchLocations[key][1:]
-
-    # Use findPath and findBestRoots to construct the DTLOR graph dictionary
-    #this finds the optimal solutions (mappings)
-    treeMin, min_cost = findBestRoots(parasiteTree, Minimums)
-    #This picks a random MPR from the optimal ones
-    MPR = findOneMPR(treeMin, eventsDict, {})  
+                    ep_best_switch = best_switch[(ep, eh, l_bottom)]
+                    O_eh2=O[(ep, eh2, l_bottom)]
+                    O_eh1=O[(ep, eh1, l_bottom)]
+                    best_switch[(ep, eh1, l_bottom)] = combine_costs([ep_best_switch, O_eh2, O_eh1])
```

Comment:
I think best_switch is computed for this way:
best_switch[(ep, eh1)] = min(best_switch[(ep, eh)], O[(ep, eh2)])
best_switch[(ep, eh2)] = min(best_switch[(ep, eh)], O[(ep, eh1)])

Reply:
That's correct. Changed!